haystack/nodes/ranker/sentence_transformers.py

```diff
@@ -90,8 +92,16 @@ def __init__(
         self.devices, _ = initialize_device_settings(devices=devices, use_cuda=use_gpu, multi_gpu=True)
 
         self.progress_bar = progress_bar
+        self.model_kwargs = model_kwargs
+        kwargs = model_kwargs if model_kwargs else {}
+        torch_dtype = extract_torch_dtype(**kwargs)
+        if torch_dtype:
+            kwargs["torch_dtype"] = torch_dtype
```

Comment:
nitpick to think about. Why not overwrite torch_dtype in the kwargs directly so you don't have to that in the caller code which might be error prone IMHO. Perhaps add a boolean overwrite to extract_torch_dtype

Reply:
To make this more explicit I'd be happy to change the function `extract_torch_dtype` to only take a torch_dtype string as input and output correct torch class instead of taking in kwargs. Would that make it more clear what's going on?

Reply:
Yeah, I think so!